designs/2018-relative-package-loading/README.md

```diff
@@ -0,0 +1,278 @@
+    },
+    "extends": ["foo"]
+}
+```
+
+```json
+{
+    "rules": {
+        "react/no-typos": "error"
+    },
+    "extends": ["foo"],
+    "plugins": ["react"]
+}
+```
+
+In the first case, only one version of `eslint-plugin-react` is loaded (the version from `eslint-config-foo`), so the `react/no-typos` reference unambiguously refers to that version of `eslint-plugin-react`.
+
+In the second case, there are two versions of `eslint-plugin-react` loaded: one from `eslint-config-foo`, and one from the end user's config. However, no ambiguity error is raised because the latter is a *direct dependency* of the end user's config. Instead, the reference `react/no-typos` resolves to the user's version of `eslint-plugin-react`. This has the potential to be confusing.
+
+This problem arises due to the special exception in the rule name resolution algorithm where if the `configScope` is empty and the given plugin name is available as a direct dependency, then the rule reference always resolves to that dependency without doing the normal ambiguity checks. The exception exists because if such a reference was considered ambiguous, it would not be possible to disambiguate it by specifying a scope.
+
+In other words, in this case it would be possible to refer to the version of the plugin from `eslint-config-foo` with `foo::react/no-typos`, but there isn't anything that could be used in place of `foo::` if the user intended to refer to the direct dependency. Instead, the unscoped `react/no-typos` reference is allowed and refers to the direct dependency rather than being considered ambiguous.
+
+One potential alternative would be to introduce a separate syntax to refer to direct dependencies of the current config file. For example, `::react/no-typos` (only a prefix of `::` with nothing before it) could refer to the direct dependency, and `foo::react/no-typos` could refer to the version from `eslint-config-foo`. The drawbacks of this alternative would be an increase in complexity and a loss of the invariant that a user can always use `pluginName/ruleName` to refer to plugins that they directly depend on, regardless of other things they have installed.
+
+As a sidenote, this example demonstrates the presence of the `plugins` field in a config is now significant, whereas previously it could sometimes be omitted without changing anything if a plugin was already being loaded from somewhere else.
+
+### Home-directory configs which depend on shareable configs will usually stop working
+
+With this change, config files in a home directory will attempt to load their shareable configs from the home directory, usually resulting in an error. (Previously, these home-directory configs would implicitly depend on having certain plugins available from whatever version of ESLint was being used to load them.) Given this constraint, we might consider simply dropping support for home-directory configs, as has been discussed separately from this proposal in the past.
```

Comment:
Agreed. This seems like a very big deal.

Reply:
Ok, it looks like there isn't much appetite for removing home-directory configs entirely, so I'll remove that note.

@nzakas: Do you mean that removing home-directory configs entirely seems like a very big deal (which is addressed by removing that suggestion from the proposal)? Or do you mean that changing how plugins are loaded from home-directory configs seems like a very big deal? Since you indicated agreement with @btmills, I'm assuming the former, but feel free to mark this conversation as unresolved if I misunderstood.